src/hotspot/share/opto/lcm.cpp

```diff
@@ -1269,7 +1269,7 @@ Node* PhaseCFG::catch_cleanup_find_cloned_def(Block *use_blk, Node *def, Block *
   if( j == def_blk->_num_succs ) {
     // Block at same level in dom-tree is not a successor.  It needs a
     // PhiNode, the PhiNode uses from the def and IT's uses need fixup.
-    Node_Array inputs = new Node_List();
+    Node_Array inputs;
```

Comment:
Urgh, looks like we had object slicing here before your changes.

Reply:
Yup, there were a few nasty things happening :sweat_smile: